src/wutil.cpp

```diff
@@ -711,6 +711,78 @@ double fish_wcstod(const wchar_t *str, wchar_t **endptr) {
     return std::wcstod(str, endptr);
 }
 
+/// Like wcstod(), but allows underscore dividers between digits in the input string.
+double fish_wcstod_underscores(const wchar_t *str, wchar_t **endptr) {
+    // OpenBSD's strtod/wcstod do not support hex.
+    static const bool allow_hex = (fish_wcstod(L"0x1", nullptr) == 1);
```

Comment:
I don't believe this helps. We *want* reading hex numbers to fail on OpenBSD. It not having hex number support is a deficiency in OpenBSD, and should be fixed there. It's a compliance issue and technically means OpenBSD doesn't fullfil our requirements (C++11/C99 require hex support).

But anyway, I don't see how this helps: Best case, it only passes the `0` from `0xdeadbeef` to wcstod, which returns 0. Then the rest of the parsing sees an `x` without following whitespace and errors out. That means this is pointless because we could just let it read up to the `x` and fail.

Worst case, it doesn't error out and now `0x5` means "0 times 5" in OpenBSD and "5" elsewhere.

We should fail on OpenBSD so it can be fixed.

Reply:
Oh, I see this is a consequence of trying to find the end point that wcstod *would* return.